### ISSUE TITLE
Requirements for test suite

### ISSUE THREAD

**Issue by daneos** [sardana-org/sardana#561]
Are there some specific requirements for running the test suite?
Even on `master` (which I suppose should pass all tests) I have 68 failed, and with Travis they are all passing.
I run tests by installing the package and running `sardanatestsuite` script.

**Comment by cmft** [comment]
Sardana testsuite needs to  have an environment configured (a Pool and MacroServer started (with sardemo)). See the [documentation](http://www.sardana-controls.org/en/latest/devel/howto_test/test_run_commands.html#run-test-suite)

Do you have it?

You can configure the Pool and MS to be used in the sardanacustomsetting file. (By default it expect demo1)

Hope this help you

**Comment by daneos** [comment]
I thought that the test suite creates the Sardana instance by itself.
However it seems to help a lot, now I have only 9 failed tests on `master`.
I guess there is some problem with `TANGO_HOST` since it tries to connect to the `10` IP address.
```
ConnectionFailed: DevFailed[
DevError[
    desc = TRANSIENT CORBA system exception: TRANSIENT_ConnectFailed
  origin = Connection::connect
  reason = API_CorbaException
severity = ERR]

DevError[
    desc = Failed to connect to database on host 10 with port 10000
  origin = Connection::connect
  reason = API_CantConnectToDatabase
severity = ERR]
]
```
I have tango host set to `10.81.0.101` so I think it's sliced on the dots for some reason.

**Comment by daneos** [comment]
It seems that using DNS name instead of IP address helps for this, but still, it would be good to be able to use IP as `TANGO_HOST` for the test suite.

**Comment by cmft** [comment]
You are right, it seems a bug. I will try to debug it. Can you say me your setup (Tango, PyTango, Taurus, and Sardana)? Could you copy the last lines output of the test suite (those which reported test failures)?

**Comment by daneos** [comment]
I'm using Tango 9.2.5a, PyTango 9.2.1, Taurus 4.1.1 and Sardana installed from the `master` branch (2.3.2).
I also found another problem, with my `bug-259` branch. I have created a pull request (#560), and I have some failed tests on Travis, but now I've run tests for this branch locally and it's passing.

Here's the output of tests for the `master` branch with IP address as `TANGO_HOST`:
```
======================================================================
ERROR: test_ascanct_macro_runs_2 (sardana.macroserver.macros.test.test_ascanct.AscanctTest)
Testing ascanct with macro_runs(macro_params=['_test_mt_1_1', '0', '10', '100', '0.1'], wait_timeout=30, meas_config=[[('_test_ct_1_1', 'software', 0)], [('_test_ct_2_1', '_test_tg_1_1', 0)]])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/macroserver/macros/test/test_ascanct.py", line 102, in macro_runs
    self.create_meas(meas_config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_ascanct_macro_runs_3 (sardana.macroserver.macros.test.test_ascanct.AscanctTest)
Testing ascanct with macro_runs(macro_params=['_test_mt_1_1', '0', '10', '100', '0.1'], wait_timeout=30, meas_config=[[('_test_ct_1_1', '_test_tg_1_1', 0), ('_test_ct_1_2', '_test_tg_1_1', 0)]])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/macroserver/macros/test/test_ascanct.py", line 102, in macro_runs
    self.create_meas(meas_config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_ascanct_macro_runs_4 (sardana.macroserver.macros.test.test_ascanct.AscanctTest)
Testing ascanct with macro_runs(macro_params=['_test_mt_1_1', '0', '10', '100', '0.1'], wait_timeout=30, meas_config=[[('_test_ct_1_1', '_test_tg_1_1', 0)]])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/macroserver/macros/test/test_ascanct.py", line 102, in macro_runs
    self.create_meas(meas_config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_ascanct_macro_stops (sardana.macroserver.macros.test.test_ascanct.AscanctTest)
Testing ascanct with macro_stops(macro_params=['_test_mt_1_1', '0', '10', '100', '0.1'], stop_delay=5, wait_timeout=20, meas_config=[[('_test_ct_1_1', '_test_tg_1_1', 0)]])
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/macroserver/macros/test/test_ascanct.py", line 163, in macro_stops
    self.create_meas(meas_config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_meas_cont_acquisition (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Synchronized acquisition with four channels from two differentcontrollers using hardware and software triggers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 256, in meas_cont_acquisition
    self.create_meas(config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_meas_cont_acquisition_2 (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Synchronized acquisition with two channels from the same controller using hardware trigger
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 256, in meas_cont_acquisition
    self.create_meas(config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_stop_meas_cont_acquisition (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Stop of the synchronized acquisition with four channels from two different controllers using hardware and software triggers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 270, in stop_meas_cont_acquisition
    self.create_meas(config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

======================================================================
ERROR: test_stop_meas_cont_acquisition_2 (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Stop of the synchronized acquisition with two channels from the same controller using software trigger
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 287, in stop_meas_cont_acquisition
    channel = PyTango.DeviceProxy(name)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 172, in __DeviceProxy__init__
    return DeviceProxy.__init_orig__(self, *args, **kwargs)
ConnectionFailed: DevFailed[
DevError[
    desc = TRANSIENT CORBA system exception: TRANSIENT_ConnectFailed
  origin = Connection::connect
  reason = API_CorbaException
severity = ERR]

DevError[
    desc = Failed to connect to database on host 10 with port 10000
  origin = Connection::connect
  reason = API_CantConnectToDatabase
severity = ERR]
]

======================================================================
ERROR: test_stop_meas_cont_acquisition_3 (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Stop of the synchronized acquisition with two channels from the same controller using hardware trigger
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 270, in stop_meas_cont_acquisition
    self.create_meas(config)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/test/test_measurementgroup.py", line 129, in create_meas
    self.meas.write_attribute('configuration', json.dumps(cfg))
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 264, in greener
    ret = submitter(fn, self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/green.py", line 100, in submit
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tango/device_proxy.py", line 1281, in __DeviceProxy__write_attribute
    return self._write_attribute(*args, **kwargs)
DevFailed: DevFailed[
DevError[
    desc = KeyError: "There is no element with full name '10:10000/triggergate/_test_tg_ctrl_1/1'"
           
  origin =   File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/tango/pool/MeasurementGroup.py", line 217, in write_Configuration
    self.measurement_group.set_configuration_from_user(cfg)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/pool/poolmeasurementgroup.py", line 434, in set_configuration_from_user
    synchronizer = pool.get_element_by_full_name(synchronizer)
  File "/usr/local/lib/python2.7/dist-packages/sardana-2.3.2-py2.7.egg/sardana/sardanacontainer.py", line 176, in get_element_by_full_name
    "There is no element with full name '%s'" % full_name)

  reason = PyDs_PythonError
severity = ERR]

DevError[
    desc = Failed to write_attribute on device mntgrp/unittest1_1/_test_mg_1, attribute configuration
  origin = DeviceProxy::write_attribute()
  reason = API_AttributeFailed
severity = ERR]
]

----------------------------------------------------------------------
Ran 180 tests in 527.372s

FAILED (errors=9, skipped=4, expected failures=1)
```

**Comment by cmft** [comment]
Thanks @danes,
The error is in the test util,  for any reason  it split to avoid the domain.
```
obj_fullname = '%s:%s/%s' % (dev.get_db_host().split('.')[0],
                                         dev.get_db_port(),
                                         dev.name())
```
I will try to fix it.

**Comment by reszelaz** [comment]
I think that this test was fixed in e953bd3. Also the original question was replied by @cmft.

@daneos I'm closing this, just reopen if you still have problems.
